nifi-extension-bundles/nifi-standard-services/nifi-database-dialect-service-api/src/main/java/org/apache/nifi/database/dialect/service/api/ColumnDefinition.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.database.dialect.service.api;
+
+import java.util.Optional;
+
+/**
+ * Database Table Column Definition
+ */
+public interface ColumnDefinition {
+    String columnName();
+
+    int dataType();
+
+    Nullable nullable();
+
+    Optional<String> defaultValue();
```

Comment:
The `defaultValue()` doesn't seem to be called from anywhere. Is it intentional?

Reply:
That's a good point. This was carried over from the current structure, but as you noted, it is not used, so I will look at removing it.